python/plugins/processing/tools/vector.py

```diff
@@ -521,6 +523,21 @@ def ogrLayerName(uri):
         regex = re.compile('(layername=)(.*)')
         r = regex.search(uri)
         return r.groups()[1]
+    elif '.gdb' in uri:
```

Comment:
Actually if you changed your "elif '.gdb' in uri:" as a simple else (and remove the existing else clause), that should be more general